EIPS/eip-5089.md

```diff
@@ -0,0 +1,391 @@
+
+MUST emit the `Redeem` event.
+
+MUST support a withdraw flow where the principal tokens are burned from `owner` directly where `owner` is `msg.sender` or `msg.sender` has [ERC-20](./eip-20.md) approval over the principal tokens of `owner`.
+ MAY support an additional flow in which the principal tokens are transferred to the principal token contract before the `withdraw` execution, and are accounted for during `withdraw`.
+
+MUST revert if all of `underlyingAmount` cannot be withdrawn (due to withdrawal limit being reached, slippage, the owner not having enough principal tokens, etc).
+
+Note that some implementations will require pre-requesting to the principal token contract before a withdrawal may be performed. Those methods should be performed separately.
+
+```yaml
+- name: withdraw
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: underlyingAmount
+      type: uint256
+    - name: receiver
+      type: address
+    - name: owner
+      type: address
+
+  outputs:
+    - name: principalAmount
+      type: uint256
+```
+
+### Events
+
+#### Redeem
+
+`from` has exchanged `principalAmount` of Principal Tokens for `underlyingAmount` of underlying, and transferred that underlying to `to`.
+
+MUST be emitted when Principal Tokens are burnt and underlying is withdrawn from the contract in the `ERC5089.redeem` method.
+
+```yaml
+- name: Redeem
+  type: event
+
+  inputs:
+    - name: from
+      indexed: true
+      type: address
+    - name: to
+      indexed: true
+      type: address
+    - name: amount
+      indexed: false
+      type: uint256
+```
+
+#### Mature
+The Principal Token has hit maturity, and a a state change has been recorded on the Principal Token contract for the first time post-maturity.
+
+```yaml
+- name: Mature
+  type: event
+
+  inputs:
+    - name: timestamp
+      indexed: true
+      type: uint256
+```
+
+## Rationale
+
+The Principal Token interface is designed to be optimized for integrators with a core minimal interface alongside optional interfaces to enable backwards compatability. Details such as accounting and management of underlying are intentionally not specified, as Principal Tokens are expected to be treated as black boxes on-chain and inspected off-chain before use.
+
+[ERC-20](./eip-20.md) is enforced as implementation details such as token approval and balance calculation directly carry over. This standardization makes Principal Tokens immediately compatible with all [ERC-20](./eip-20.md) use cases in addition to ERC-5089.
+
+All principal tokens are redeemable upon maturity, with the only variance being whether further yield is generated post-maturity. Given the ubiquity of redemption, the presence of `redeem` allows integrators to purchase Principal Tokens on an open market, and them later redeem them for a fixed-yield solely knowing the address of the Principal Token itself.
```

Comment:
This assumption only allows this standard to work in the case principal tokens are redeemable for the principal + yield. There are some systems where it has to be redeemed in tandem with the yield tokens.

Do you have any suggestion for covering that?

Reply:
I've been asking around and I haven't found a system like the one that you describe that would be relevant, would you have any more detail?